hexa/pipelines/tests/test_schema.py

```diff
@@ -516,3 +539,225 @@ def test_delete_pipeline_version(self):
+        self.WORKSPACE1_MEMBERSHIP_2.delete()
+
+        r = self.run_query(
+            """
+            query pipelineByCode($code: String!, $workspaceSlug: String!) {
+                pipelineByCode(code: $code, workspaceSlug: $workspaceSlug) {
+                    recipients {
+                      user {
+                        displayName
+                      }
+                    }
+                }
+            }
+        """,
+            {
+                "code": pipeline.code,
+                "workspaceSlug": self.WS1.slug,
+            },
+        )
+
+        self.assertEqual(
+            {"recipients": []},
+            r["data"]["pipelineByCode"],
+        )
+
+    @test.override_settings(NEW_FRONT_DOMAIN="localhost:3000")
+    def test_mail_run_recipients_manual_trigger(self):
+        self.client.force_login(self.USER_ROOT)
+        self.test_pipeline_new_run()
+
+        pipeline = Pipeline.objects.filter_for_user(user=self.USER_ROOT).first()
+        version = PipelineVersion.objects.get(pipeline=pipeline, number=1)
+        run = pipeline.run(
+            user=self.USER_ROOT,
+            pipeline_version=version,
+            trigger_mode=PipelineRunTrigger.MANUAL,
+            config={},
+            send_mail_notifications=True,
+        )
+
+        with patch("hexa.pipelines.utils.send_mail") as send_mail:
+            mail_run_recipients(run)
+            send_mail.assert_called_with(
+                title=f"Run report of {pipeline.code}({run.state.name})",
+                template_name="pipelines/mails/run_report",
+                template_variables={
+                    "pipeline_code": pipeline.code,
+                    "status": run.status.name,
+                    "executed_at": run.execution_date,
+                    "duration": int(run.duration.total_seconds())
+                    if run.duration is not None
+                    else 0,
+                    "run_url": f"https://localhost:3000/workspaces/{pipeline.workspace.slug}/pipelines/{run.pipeline.code}/runs/{run.id}",
+                },
+                recipient_list=[self.USER_ROOT.email],
+            )
+
+    @test.override_settings(NEW_FRONT_DOMAIN="localhost:3000")
+    def test_mail_run_recipients_scheduled_trigger(self):
+        self.client.force_login(self.USER_ROOT)
+        self.test_pipeline_new_run()
+
+        pipeline = Pipeline.objects.filter_for_user(user=self.USER_ROOT).first()
+        version = PipelineVersion.objects.get(pipeline=pipeline, number=1)
+        PipelineRecipient.objects.create(pipeline=pipeline, user=self.USER_ROOT)
+        PipelineRecipient.objects.create(pipeline=pipeline, user=self.USER_LAMBDA)
+
+        run = pipeline.run(
+            user=self.USER_ROOT,
+            pipeline_version=version,
+            trigger_mode=PipelineRunTrigger.SCHEDULED,
+            config={},
+            send_mail_notifications=True,
+        )
+
+        with patch("hexa.pipelines.utils.send_mail") as send_mail:
```

Comment:
No need to patch `send_mail`, it's builtin in Django. Check `def test_invite_workspace_member()`